packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -235,66 +282,104 @@ export class SummarizerNode implements IRootSummarizerNode {
 		readAndParseBlob: ReadAndParseBlob,
 		correlatedSummaryLogger: ITelemetryLogger,
 	): Promise<RefreshSummaryResult> {
-		this.defaultLogger.sendTelemetryEvent({
-			eventName: "refreshLatestSummary_start",
+		const eventProps: {
+			proposalHandle: string | undefined;
+			summaryRefSeq: number;
+			referenceSequenceNumber: number;
+			latestSummaryUpdated?: boolean;
+			wasSummaryTracked?: boolean;
+		} = {
 			proposalHandle,
-			referenceSequenceNumber: this.referenceSequenceNumber,
 			summaryRefSeq,
-		});
+			referenceSequenceNumber: this.referenceSequenceNumber,
+		};
+		return PerformanceEvent.timedExecAsync(
+			this.logger,
+			{
+				eventName: "refreshLatestSummary",
+				...eventProps,
+			},
+			async (event) => {
+				// Refresh latest summary should not happen while a summary is in progress. If it does, it can result
+				// in inconsistent state, so, we should not continue;
+				if (this.isSummaryInProgress()) {
+					throw new LoggingError("UnexpectedRefreshDuringSummarize", {
```

Comment:
@NicholasCouri I added this validation here. Do you think I should wait until https://dev.azure.com/fluidframework/internal/_workitems/edit/3385 is fixed?